providers/cloudflare.go

```diff
@@ -0,0 +1,204 @@
+package providers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/crackcomm/cloudflare"
+	"github.com/rancher/external-dns/dns"
+	"golang.org/x/net/context"
+)
+
+type CloudflareHandler struct {
+	client *cloudflare.Client
+	zone   *cloudflare.Zone
+	ctx    context.Context
+	root   string
+}
+
+func init() {
+	email := os.Getenv("CLOUDFLARE_EMAIL")
+	if len(email) == 0 {
+		logrus.Info("CLOUDFLARE_EMAIL is not set, skipping init of CloudFlare provider")
```

Comment:
I doubt that the provider name would change, but do you think it would make sense to call getName() instead of hardcoding the provider name?